Register a hypothesis CI profile?

Hello, our Nixpkgs CI has encountered the following error on `aarch64-linux`:

```
=================================== FAILURES ===================================
_____________________________ test_table_variable ______________________________
  + Exception Group Traceback (most recent call last):
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/runner.py", line 341, in from_call
  |     result: TResult | None = func()
  |                              ~~~~^^
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/runner.py", line 242, in <lambda>
  |     lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  |             ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_hooks.py", line 513, in __call__
  |     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  |            ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_manager.py", line 120, in _hookexec
  |     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  |            ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_callers.py", line 182, in _multicall
  |     return outcome.get_result()
  |            ~~~~~~~~~~~~~~~~~~^^
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_result.py", line 100, in get_result
  |     raise exc.with_traceback(exc.__traceback__)
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_callers.py", line 167, in _multicall
  |     teardown.throw(outcome._exception)
  |     ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/threadexception.py", line 92, in pytest_runtest_call
  |     yield from thread_exception_runtest_hook()
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/threadexception.py", line 68, in thread_exception_runtest_hook
  |     yield
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_callers.py", line 167, in _multicall
  |     teardown.throw(outcome._exception)
  |     ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/unraisableexception.py", line 95, in pytest_runtest_call
  |     yield from unraisable_exception_runtest_hook()
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/unraisableexception.py", line 70, in unraisable_exception_runtest_hook
  |     yield
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_callers.py", line 167, in _multicall
  |     teardown.throw(outcome._exception)
  |     ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/logging.py", line 846, in pytest_runtest_call
  |     yield from self._runtest_for(item, "call")
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/logging.py", line 829, in _runtest_for
  |     yield
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_callers.py", line 167, in _multicall
  |     teardown.throw(outcome._exception)
  |     ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/capture.py", line 898, in pytest_runtest_call
  |     return (yield)
  |             ^^^^^
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_callers.py", line 167, in _multicall
  |     teardown.throw(outcome._exception)
  |     ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/skipping.py", line 257, in pytest_runtest_call
  |     return (yield)
  |             ^^^^^
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_callers.py", line 103, in _multicall
  |     res = hook_impl.function(*args)
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/runner.py", line 174, in pytest_runtest_call
  |     item.runtest()
  |     ~~~~~~~~~~~~^^
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/python.py", line 1627, in runtest
  |     self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_hooks.py", line 513, in __call__
  |     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  |            ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_manager.py", line 120, in _hookexec
  |     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  |            ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
  |     raise exception.with_traceback(exception.__traceback__)
  |   File "/nix/store/yn8wwxzjya9askj7z6r0ifcl75m2f0ps-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_callers.py", line 103, in _multicall
  |     res = hook_impl.function(*args)
  |   File "/nix/store/lcxwcpg47mcmz96c7dd2qswagb6vz6kv-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/python.py", line 159, in pytest_pyfunc_call
  |     result = testfunction(**testargs)
  |   File "/build/source/tests/html/table_test.py", line 29, in test_table_variable
  |     @settings()
  |     
  |   File "/nix/store/a54092axlvz4cvcbfjayjg9azbgh5lnb-python3.13-hypothesis-6.130.12/lib/python3.13/site-packages/hypothesis/core.py", line 1837, in wrapped_test
  |     raise the_error_hypothesis_found
  | hypothesis.errors.FlakyFailure: Hypothesis test_table_variable(var=<scipp.Variable> (: 1)      int32  [dimensionless]  [0]) produces unreliable results: Falsified on the first call but did not on a subsequent one (1 sub-exception)
  | Falsifying example: test_table_variable(
  |     var=functools.partial(array, dims=[''], unit='one')(
  |         values=array([0], dtype=int32),
  |         variances=None,
  |     ),
  | )
  | Unreliable test timings! On an initial run, this test took 3373.09ms, which exceeded the deadline of 1000.00ms, but on a subsequent run it took 1.61 ms, which did not. If you expect this sort of variability in your test timings, consider turning deadlines off for this test by setting deadline=None.
  | 
  | You can reproduce this example by temporarily adding @reproduce_failure('6.130.12', b'AIBBAUEAQQBBAABBAA==') as a decorator on your test case
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/nix/store/a54092axlvz4cvcbfjayjg9azbgh5lnb-python3.13-hypothesis-6.130.12/lib/python3.13/site-packages/hypothesis/core.py", line 1085, in _execute_once_for_engine
    |     result = self.execute_once(data)
    |   File "/nix/store/a54092axlvz4cvcbfjayjg9azbgh5lnb-python3.13-hypothesis-6.130.12/lib/python3.13/site-packages/hypothesis/core.py", line 1022, in execute_once
    |     result = self.test_runner(data, run)
    |   File "/nix/store/a54092axlvz4cvcbfjayjg9azbgh5lnb-python3.13-hypothesis-6.130.12/lib/python3.13/site-packages/hypothesis/core.py", line 729, in default_executor
    |     return function(data)
    |   File "/nix/store/a54092axlvz4cvcbfjayjg9azbgh5lnb-python3.13-hypothesis-6.130.12/lib/python3.13/site-packages/hypothesis/core.py", line 996, in run
    |     return test(*args, **kwargs)
    |   File "/build/source/tests/html/table_test.py", line 29, in test_table_variable
    |     @settings()
    |     
    |   File "/nix/store/a54092axlvz4cvcbfjayjg9azbgh5lnb-python3.13-hypothesis-6.130.12/lib/python3.13/site-packages/hypothesis/core.py", line 926, in test
    |     raise DeadlineExceeded(
    |         datetime.timedelta(seconds=runtime), self.settings.deadline
    |     )
    | hypothesis.errors.DeadlineExceeded: Test took 3373.09ms, which exceeds the deadline of 1000.00ms
    +------------------------------------
=========================== short test summary info ============================
SKIPPED [1] tests/bins_test.py:599: Need fix in Variable::setSlice
SKIPPED [1] tests/datetime_test.py:273: Requires std::chrono from C++20
FAILED tests/html/table_test.py::test_table_variable - hypothesis.errors.FlakyFailure: Hypothesis test_table_variable(var=<scipp.Variable> (: 1)      int32  [dimensionless]  [0]) produces unreliable results: Falsified on the first call but did not on a subsequent one (1 sub-exception)
Falsifying example: test_table_variable(
    var=functools.partial(array, dims=[''], unit='one')(
        values=array([0], dtype=int32),
        variances=None,
    ),
)
Unreliable test timings! On an initial run, this test took 3373.09ms, which exceeded the deadline of 1000.00ms, but on a subsequent run it took 1.61 ms, which did not. If you expect this sort of variability in your test timings, consider turning deadlines off for this test by setting deadline=None.
You can reproduce this example by temporarily adding @reproduce_failure('6.130.12', b'AIBBAUEAQQBBAABBAA==') as a decorator on your test case
============ 1 failed, 5193 passed, 2 skipped in 100.69s (0:01:40) =============
```

I think this is the same error as I encountered in the past with astropy:

- https://github.com/astropy/astropy/issues/17649

Would it be possible for you to add something like the following to your `conftest.py`? I would have opened a PR by myself but it is trivial and I found it a bit more natural to explain the situation in an issue.

```python
hypothesis.settings.register_profile(
    "ci",
    deadline=None,
    print_blob=True,
    derandomize=True,
)
```

Since the error I got is very hard to reproduce, I'm not sure I'd be able to help much with testing, besides reporting you whether running pytest with `--hypothesis-profile=ci` is not raising an error.

--- a/.github/workflows/nightly-deps-test.yml
+++ b/.github/workflows/nightly-deps-test.yml
@@ -56,4 +56,4 @@ jobs:
       - run: cmake --preset ${{ matrix.variant.cmake-preset }}
       - run: cmake --build --preset build
       - run: ctest --preset test
-      - run: tox -e nightlydeps -vvv
+      - run: tox -e nightlydeps -vvv -- --hypothesis-profile=ci

--- a/.github/workflows/pr_and_main.yml
+++ b/.github/workflows/pr_and_main.yml
@@ -79,7 +79,7 @@ jobs:
       - run: ctest --preset test
 
       - run: tox -e import-minimal
-      - run: tox -e test
+      - run: tox -e test -- --hypothesis-profile=ci
 
       - run: tox -e mypy
         if: ${{ contains(matrix.variant.os, 'ubuntu') }}

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -4,6 +4,7 @@
 import os
 from typing import Any
 
+import hypothesis
 import pytest
 
 import scipp as sc
@@ -20,3 +21,14 @@ def pytest_assertrepr_compare(op: str, left: Any, right: Any) -> list[str] | Non
     if isinstance(left, sc.DType) or isinstance(right, sc.DType):
         return [f'{left!r} {op} {right!r}']
     return None
+
+
+hypothesis.settings.register_profile(
+    "ci",
+    # Disable the time limit, see
+    # https://github.com/scipp/scipp/issues/3721
+    # https://github.com/astropy/astropy/issues/17649
+    deadline=None,
+    # Print the random seed blob for reproducibility:
+    print_blob=True,
+)

--- a/tox.ini
+++ b/tox.ini
@@ -13,7 +13,7 @@ deps = -r requirements/test.txt
 setenv =
   PYTHONPATH = {toxinidir}{/}install
   JUPYTER_PLATFORM_DIRS = 1
-commands = python -X PYTHONWARNDEFAULTENCODING -m pytest -n auto -v {posargs:tests}
+commands = python -X PYTHONWARNDEFAULTENCODING -m pytest -n auto -v {posargs:}
 
 [testenv:import-minimal]
 description = Test if scipp can be imported when installed with only base dependencies
@@ -132,7 +132,7 @@ setenv =
   JUPYTER_PLATFORM_DIRS = 1
   PIP_INDEX_URL = {env:PIP_INDEX_URL:https://pypi.anaconda.org/scientific-python-nightly-wheels/simple}
   PIP_EXTRA_INDEX_URL = {env:PIP_EXTRA_INDEX_URL:https://pypi.org/simple}
-commands = python -X PYTHONWARNDEFAULTENCODING -m pytest -n auto -v {posargs:tests}
+commands = python -X PYTHONWARNDEFAULTENCODING -m pytest -n auto -v {posargs:}
 
 [testenv:mypy]
 description = Type checking (mypy)

